pkg/kube/vault/vault.go

```diff
@@ -121,19 +144,90 @@ func SystemVaultNameForCluster(clusterName string) string {
 	return cluster.ShortNameN(fullName, 22)
 }
 
-// CreateVault creates a new vault backed by GCP KMS and storage
-func CreateVault(kubeClient kubernetes.Interface, vaultOperatorClient versioned.Interface, name string, ns string,
+// CreateGKEVault creates a new vault backed by GCP KMS and storage
+func CreateGKEVault(kubeClient kubernetes.Interface, vaultOperatorClient versioned.Interface, name string, ns string,
 	gcpServiceAccountSecretName string, gcpConfig *GCPConfig, authServiceAccount string,
 	authServiceAccountNamespace string, secretsPathPrefix string) error {
 
+	vault, err := InitializeVault(kubeClient, name, ns, authServiceAccount, authServiceAccountNamespace, secretsPathPrefix)
+	if err != nil {
+		return err
+	}
+
+	vault.Spec.Config["storage"] = Storage{
+		GCS: GCSConfig{
+			Bucket:    gcpConfig.GcsBucket,
+			HaEnabled: "true",
+		},
+	}
+	vault.Spec.UnsealConfig =  v1alpha1.UnsealConfig{
+		Google: &v1alpha1.GoogleUnsealConfig{
+			KMSKeyRing:    gcpConfig.KmsKeyring,
+			KMSCryptoKey:  gcpConfig.KmsKey,
+			KMSLocation:   gcpConfig.KmsLocation,
+			KMSProject:    gcpConfig.ProjectId,
+			StorageBucket: gcpConfig.GcsBucket,
+		},
+	}
+	vault.Spec.CredentialsConfig = v1alpha1.CredentialsConfig{
+		Env:        gcpServiceAccountEnv,
+		Path:       gcpServiceAccountPath,
+		SecretName: gcpServiceAccountSecretName,
+	}
+
+
+	_, err = vaultOperatorClient.VaultV1alpha1().Vaults(ns).Create(vault)
+	return err
+}
+
+// CreateAWSVault creates a new vault backed by AWS KMS and DynamoDB storage
+func CreateAWSVault(kubeClient kubernetes.Interface, vaultOperatorClient versioned.Interface, name string, ns string,
+	awsConfig *AWSConfig, authServiceAccount string, authServiceAccountNamespace string, secretsPathPrefix string) error {
+
+	vault, err := InitializeVault(kubeClient, name, ns, authServiceAccount, authServiceAccountNamespace, secretsPathPrefix)
+	if err != nil {
+		return err
```

Comment:
wrap err

Reply:
Is it really needed when all errors returned from InitializeVault are wrapped?